rviz crashes with pre-built container

When running `rocker --x11 -- miguelprada/tracebot_mockup:noetic roslaunch tracebot_mockup_description view_tracebot_mockup.launch`, the `joint_state_publisher_gui`opens but rviz crashes at startup.

I've tested on Ubuntu 16.04 with docker 19.03.14, rocker 0.2.3 and nvidia-455 and on 20.04 with docker 20.10.7, same rocker version and nvidia-460. The latter has a clean (nvidia-)docker installation.

The rocker examples (rviz and gazebo) are working. I've also been able to get `tracebot_mockup` to work by manually installing everything, starting from the ROS Melodic base image and then following your instructions. So I assume this is rather an issue with the container.

If someone is able to reproduce this error, please verify. If it is working for you, please check if there are any additional flags or settings missing from the documentation.

PS: The rocker documentation misses the information that `python3-distutils` needs to be installed beforehand, if you run into any trouble there.

For me, if I run `rocker --pull --x11 --nvidia -- miguelprada/tracebot_mockup:noetic` I get:

```
output:  standard_init_linux.go:211: exec user process caused "permission denied"

/tmp/detect_os failed:
> standard_init_linux.go:211: exec user process caused "permission denied"
WARNING unable to detect os for base image 'miguelprada/tracebot_mockup:noetic', maybe the base image does not exist

```

and if I run `rocker --pull --x11 -- miguelprada/tracebot_mockup:noetic` I can get a prompt, but running `roslaunch tracebot_mockup_description view_tracebot_mockup.launch` inside the container gives me 
```
[ERROR] [1624615021.429683022]: Unable to create the rendering window after 100 tries.
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
[rviz-4] process has died [pid 57, exit code -6, cmd /opt/ros/noetic/lib/rviz/rviz -d /opt/ros/tracebot_mockup-noetic/share/tracebot_mockup_description/config/tracebot_mockup.rviz __name:=rviz __log:=/root/.ros/log/af36db76-d59b-11eb-85a7-0242ac110002/rviz-4.log].
log file: /root/.ros/log/af36db76-d59b-11eb-85a7-0242ac110002/rviz-4*.log
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast

```

--- a/README.md
+++ b/README.md
@@ -106,4 +106,10 @@ The mockup with default parameters can be run using rocker with:
 rocker --x11 -- miguelprada/tracebot_mockup:noetic roslaunch tracebot_mockup_description view_tracebot_mockup.launch
 ```
 
+Note that if your system runs NVidia graphics, you may need to use:
+
+```bash
+rocker --x11 --nvidia -- miguelprada/tracebot_mockup:noetic roslaunch tracebot_mockup_description view_tracebot_mockup.launch
+```
+
 Alternatively, instructions to enable X11 forwarding in docker in different ways can be found in [this page](http://wiki.ros.org/docker/Tutorials/GUI).

